leaflet/admin.py

```diff
@@ -38,6 +38,23 @@ def formfield_for_dbfield(self, db_field, **kwargs):
         if is_editable:
             kwargs.pop('request', None)  # unsupported for form field
             # Setting the widget with the newly defined widget.
+            if 'widget' in kwargs:
+                w = kwargs['widget']
+                # If the widget is already a LeafletWidget of some kind
+                # Then use it rather than a blank one.
+                if issubclass(w, LeafletWidget):
+                    class LeafletMap(w):
```

Comment:
You should add an optionnal parameter to `_get_map_widget` with the widget to avoid to duplicate the code. This could also simplify it:

```python
if 'widget' in kwargs and issubclass(kwargs['widget'], LeafletWidget):
    widget = kwargs['widget']
else:
    widget = None
kwargs['widget'] = self._get_map_widget(db_field, widget)
```

Reply:
I made the argument mandatory, since it's only called from that function and it's private (at least in theory!).